### PR TITLE
Restore and fix disabled tests

### PR DESCRIPTION
**Pull request by S2Ler**
Fixes disabled before tests:
- Better Navigator reset method
- Hard retain in new reroute code
- An attempted fix to 'tripPerWaypoint' test

### CODE REVIEW

File: Tests/MapboxCoreNavigationTests/BillingHandlerTests.swift [hunk 1]
```diff
@@ -590,34 +590,36 @@ final class BillingHandlerUnitTests: TestCase {
...
 
-            let waypoints = coordinates.map({ Waypoint(coordinate: $0) })
 
             return NavigationRouteOptions(waypoints: waypoints)
         }
         
         let route = Fixture.route(from: "route-with-10-legs", options: routeOptions)
         
         autoreleasepool {
-            let replayLocations = Fixture.generateTrace(for: route).shiftedToPresent().qualified()
```
Comment (jill-cardamon) [inline Tests/MapboxCoreNavigationTests/BillingHandlerTests.swift:617]: How can we drop `qualified()` here? Don't we need the route to be qualified for nav native?

**Review by jill-cardamon — APPROVED** [review]
lgtm otherwise :)

File: Tests/MapboxCoreNavigationTests/BillingHandlerTests.swift (hunk 1 above)
Comment (S2Ler) [reply Tests/MapboxCoreNavigationTests/BillingHandlerTests.swift:617]: It works without this `qualified` function. Maybe it was needed at some point in the past or needed for some usecases?

Comment (S2Ler) [reply Tests/MapboxCoreNavigationTests/BillingHandlerTests.swift:617]: I will get back to this if we continue to see issues with this test.

**Pull request merged**